src/v/cluster/partition_balancer_backend.h

```diff
@@ -91,13 +92,16 @@ class partition_balancer_backend {
     consensus_ptr _raft0;
 
     controller_stm& _controller_stm;
+    features::feature_table& _feature_table;
     partition_balancer_state& _state;
     health_monitor_backend& _health_monitor;
     partition_allocator& _partition_allocator;
     topics_frontend& _topics_frontend;
     members_frontend& _members_frontend;
 
-    config::binding<model::partition_autobalancing_mode> _mode;
+    features::sanctioning_binding<
+      config::enum_property<model::partition_autobalancing_mode>>
+      _mode;
```

Comment:
observation: kind of annoying that the whole property type has to appear here. maybe a followup could get it down to the underlying value type.

nothing to do at this time IMO.